File: Java/23_Math/LC_1180_CountSubstringsWithOnlyOneDistinctLetter.java
class Solution {
    public int countLetters(String s) {
        int res = 0;
        int count = 1;
        for(int i = 1; i < s.length(); i++){
            if(s.charAt(i) == s.charAt(i - 1)){
                count++;
            }
            else{
                res += count * (count + 1) / 2;
                count = 1;
            }
        }

        res += count * (count + 1) / 2;
        return res;
    }
}

File: JavaScript/23_Math/LC_1180_CountSubstringsWithOnlyOneDistinctLetter.js
/**
 * @param {string} s
 * @return {number}
 */
var countLetters = function(s) {
    var res = 0;
    var count = 1;
    for(i = 1; i < s.length; i++){
        if(s[i] == s[i - 1]){
            count++;
        }
        else{
            res += count * (count + 1) / 2;
            count = 1;
        }
    }

    res += count * (count + 1) / 2;
    return res;
};